src/pages/permissions/CreateIdentityModal.tsx

```diff
@@ -0,0 +1,98 @@
+import { FC, useState } from "react";
+import { ActionButton, Input, Modal } from "@canonical/react-components";
+import { useFormik } from "formik";
+import Loader from "components/Loader";
+import { fetchIdentityDetails } from "api/auth-identities";
+import { base64EncodeTokenDetails } from "util/helpers";
+
+interface Props {
+  onClose: () => void;
+}
+
+const CreateIdentityModal: FC<Props> = ({ onClose }) => {
+  const [isLoadingToken, setIsLoadingToken] = useState(false);
+  const [token, setToken] = useState<string | null>(null);
+
+  const handleClose = () => {
+    onClose();
+    window.location.reload();
+  };
+
+  const formik = useFormik({
+    initialValues: {
+      clientName: "",
+    },
+    onSubmit: (values) => {
+      setIsLoadingToken(true);
+
+      fetchIdentityDetails(values.clientName)
+        .then((tokenPreResponse) => {
+          const tempToken = base64EncodeTokenDetails(tokenPreResponse);
+
+          setIsLoadingToken(false);
+          setToken(tempToken);
+        })
+        .catch((err) => {
+          setIsLoadingToken(false);
+          console.error("Error fetching token:", err);
```

Comment:
please remove the console log and use toast notification for error and success messages

Reply:
Will we need a success message if the token is being presented right in front of us and we are already in a modal? I note this because if one exits the modal after generating a token by pressing refresh, we wouldn't see the toast notification anyway. Also a token would only be generated if the operation was successful.

Reply:
that's a interesting one, I think perhaps have some copy in the modal to indicate that an TLS identity was created. I'm not entirely confident that is the correct design though, probably a good topic to discuss during standup. The toast notification would be somewhat hidden when the modal is open anyway...